### ISSUE TITLE
job submission app not uploading logs

### ISSUE THREAD

**Issue by emlyn** [Azure/aztk#393]
I'm submitting a job with `aztk spark job submit`. When it completes I try to get the logs with `aztk spark job get-app-logs --id xxx --name yyy`, but I just get an error saying:
```
Logs not found in your storage account. They were either deleted or never existed.
```
Indeed the logs don't appear to be in the storage account, which makes it quite hard to debug failures. I'm pretty sure this used to work when I first tried using job submission.

Let me know if you need any extra information to help reproduce it.

**Comment by paselem** [comment]
Quick question @emlyn - did you happen to delete the job?

**Comment by emlyn** [comment]
@paselem no, I didn't delete it. It still appears in the job list, and I can still get the info with `aztk spark job get`, `list-apps` and `get-app`.

**Comment by jafreck** [comment]
What was the exit code for the application?

**Comment by emlyn** [comment]
It had an exit code of 1

**Comment by jafreck** [comment]
Could you give steps to reproduce?

**Comment by jafreck** [comment]
@emlyn I put out a patch (#395) that might fix this issue. I can't reproduce the error yet -- so I can't be sure that your particular issue is fixed.

**Comment by jafreck** [comment]
reopening this until it's confirmed fixed

**Comment by emlyn** [comment]
That fix didn't help, but I managed to grab some files off the cluster before it was shut down, and found this:
```
Traceback (most recent call last):
File "/mnt/batch/tasks/startup/wd/submit.py", line 188, in <module>
return_code = recieve_submit_request(os.path.join(os.environ['AZ_BATCH_TASK_WORKING_DIR'], 'application.yaml'))
File "/mnt/batch/tasks/startup/wd/submit.py", line 165, in recieve_submit_request
return_code = subprocess.call(cmd.to_str(), shell=True)
File "/mnt/batch/tasks/startup/wd/command_builder.py", line 55, in to_str
return " ".join(cmd)
TypeError: sequence item 12: expected str instance, int found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/mnt/batch/tasks/startup/wd/submit.py", line 190, in <module>
upload_error_log(e, os.path.join(os.environ['AZ_BATCH_TASK_WORKING_DIR'], 'application.yaml'))
File "/mnt/batch/tasks/startup/wd/submit.py", line 175, in upload_error_log
error_log.write(error)
TypeError: write() argument must be str, not TypeError
```
Looks like the command line parameters are not all being interpreted as strings, and it's failing while building the command line to execute (I think yaml will convert anything that looks like a number to an int)?

**Comment by jafreck** [comment]
Thanks for the feedback. The error you were experiencing should now be fixed. In your storage account, you should be able to see a file called error.log in a blob container with the same name as your Job.

Could you run again and share the content of that error log? Should help with debugging this further.

I was unfortunately still not able to reproduce from your example, My Spark program crashed with bad arguments, but the logs were still uploaded as expected.

**Comment by emlyn** [comment]
Thanks for looking into this.
With the latest commit I no longer get the error saying "Logs not found in your storage account. They were either deleted or never existed.", but I just get blank output (even though the application does generate output, both to stdout and stderr). I found the err.log in the container, next to the output.log, but both of them are empty zero byte files.
After a bit more digging, I found that it's not the app arguments that are causing the problem, but the arguments to Spark: In my application config in job.yaml, I have:
```yaml
        driver_cores: 2
        executor_cores: 2
```
and this seems to be what's causing the error: when I remove those lines the output gets written again.

**Comment by jafreck** [comment]
I was able to reproduce the issue, now. It does seem to be caused by `driver_cores` and `exector_cores`.

Thanks for giving the steps to reproduce, I will fix this as soon as possible.

**Comment by jafreck** [comment]
@emlyn I believe this is now resolved after merging #399. Can you confirm?

**Comment by emlyn** [comment]
Yes, that seems to have fixed it. Thanks!
